part-3/app/views/users/login.erb

```diff
@@ -0,0 +1,14 @@
+<% if @form_error %>
+  <%= @form_error %>
+<% end %>
```

Comment:
instance variables are set to `nil` by default so you could reduce these 3 lines to just this:
```ruby
<%= @form_error %>
```